OpenRA.Mods.Common/Warheads/InstantKillWarhead.cs

```diff
@@ -0,0 +1,40 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2017 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Warheads
+{
+	[Desc("Cannot be blocked by damage modifiers.")]
+	public class InstantKillWarhead : IWarhead
+	{
+		public int Delay { get { return 0; } }
+
+		public void DoImpact(Target target, Actor firedBy, IEnumerable<int> damageModifiers)
+		{
+			var victim = target.Actor;
+			victim.Kill(firedBy);
```

Comment:
Any reason to store `target.Actor` in a local variable? Also, don't we need to check if the type of the Target is an actor in the first place?

Reply:
This should also have a damagetype list.

Reply:
> This should also have a damagetype list.
I had one before, but the immediate kill made the victim not last long enough to let it play it's animation.

Reply:
Is the issue here that the wall kills the flameguy actor?  If that is the case, then they should not be targetable by this warhead.

Reply:
It just removes everything directly from the world. I removed the `InstantKillWarhead` for now.